server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -362,6 +360,12 @@ public DesiredBalance compute(
         return new DesiredBalance(lastConvergedIndex, assignments);
     }
 
+    private static void collectShardAssignments(RoutingNodes routingNodes, HashMap<ShardId, ShardAssignment> assignments) {
```

Comment:
I think it is better to create and return the assignments map within this method rather then modify the argument

Reply:
The map is mutated later, that's why I left it in the parent scope.

Reply:
Did this after all :) because of the map pre-sizing change suggested by @arteam :)